text/0000-build-std/2-history.md

```diff
@@ -0,0 +1,535 @@
+   repositories to ensure there is no breakage. `rust-toolstate` could be used
+   but is not very good. Alternatively, Cargo could become a [JOSH] subtree of
+   [rust-lang/rust].
+
+   [wg-cargo-std-aware#86] proposes that the initial set of targets supported by
+   build-std be limited at first to further reduce scope and limit exposure to
+   the trickier issues.
+
+   [wg-cargo-std-aware#88] reports that `cargo doc -Zbuild-std` doesn't generate
+   links to the standard library. Cargo doesn't think the standard library comes
+   from crates.io, and bootstrap isn't involved to pass
+   `-Zcrate-attr="doc(html_root_url=..)"` like in the pre-built standard
+   library.
+
+   [wg-cargo-std-aware#90] asks how `restricted_std` should apply to custom
+   targets. `restricted_std` is triggered based on the `target_os` value, which
+   means it will apply for some custom targets but not others. build-std needs
+   to determine what guarantees are desirable/expected. Current implementation
+   wants slightly-modified-from-default target specs to be accepted and
+   completely new target specs to hit `restricted_std`.
+
+   [wg-cargo-std-aware#92] suggests that some targets could be made "unstable"
+   and as such only support build-std on nightly. This forces users of those
+   targets to use nightly where they will receive more frequent fixes for their
+   target. It would also permit more experimentation with build-std while
+   enabling stabilisation for mainstream targets.
+
+5. **Implementation considerations**
+   These won't be discussed in this summary, see [the implementation summary][implementation-summary]
+   or [the relevant section of the literature review for more detail][appendix-ii-impl]
+
+6. **Bugs in the compiler or standard library**
+   These aren't especially relevant to this summary, see [the relevant section
+   of the literature review for more detail][appendix-ii-bugs]
+
+7. **Cargo feature requests narrowly applied to build-std**
+   These aren't especially relevant to this summary, see [the relevant section
+   of the literature review for more detail][appendix-ii-cargo-feats]
+
+Since around 2020, activity in the [wg-cargo-std-aware] repository largely
+trailed off and there have not been any significant developments related to
+build-std since.
+
+### Implementation summary
+[implementation-summary]: #implementation-summary
+
+*An exhaustive review of implementation-related issues, pull requests and
+discussions can be found in
+[the relevant section of the literature review][appendix-ii-impl].*
+
+There has been an unstable and experimental implementation of build-std in Cargo
+since August 2019 ([wg-cargo-std-aware#10]/[cargo#7216]).
+
+[cargo#7216] added the [`-Zbuild-std`][build-std] flag to Cargo. `-Zbuild-std`
+re-builds the standard library crates which rustc then uses instead of the
+pre-built standard library from the sysroot.
+
+`-Zbuild-std` builds `std` by default. `test` is also built if `std` is being
```

Comment:
To clarify a minor point here, the default is dynamic. If `rustc` indicates that the target does not support `std`, then it only builds core. It uses the `std` field from `--print=target-spec-json` (added in https://github.com/rust-lang/rust/pull/122305).